data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsQueueUrl.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class SqsQueueUrl {
+    private final String accountId;
```

Comment:
Is the intent of this class only to get accountId ? I see we have validation for the path. Can we get ARN of the sqs endpoint and get accountId from that ?